### PR TITLE
[docs] Update dbt docs on fetching col metadata to use deferred fetch

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Updates the docs recommendation for fetching column metadata with dbt to suggest using the deferred fetch instead of fetching in-line with dbt execution.

May still want to include old docs in a collapsed section for users who are on the older APIs, right now removes them.

### CODE REVIEW

**Comment by benpankow** [comment]
* **#22950** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22950?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/integrations/dbt/reference.mdx [hunk 1]
```diff
@@ -763,57 +763,37 @@ Once your dbt models have been materialized, you can view the row count data in
...
-    experimental feature.{" "}
+    Emitting column-level metadata for dbt models is currently an experimental
+    feature.
   </strong>{" "}
-  To use this feature, you'll need to be on at least <code>dagster>=1.7.0</code>{" "}
-  and
-  <code>dagster-dbt>=0.23.0</code>.
+  To use this feature, you'll need to be on at least{" "}
+  <code>dagster>=0.17.9</code> and
+  <code>dagster-dbt>=0.23.9</code>.
```
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:771]: We should wait to land this for `1.8` so this constraint will be easier to read.

```suggestion
  <code>dagster>=1.8.0</code> and
  <code>dagster-dbt>=0.24.0</code>.
```

File: docs/content/integrations/dbt/reference.mdx [hunk 2]
```diff
@@ -763,57 +763,37 @@ Once your dbt models have been materialized, you can view the row count data in
...
-  +post-hook:
-    - "{{ dagster.log_column_level_metadata(enable_parent_relation_metadata_collection=false) }}"
+@dbt_assets(
+    manifest=manifest_path,
+)
+def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    yield from (
+        dbt.cli(["build"], context=context)
+        .stream()
+        .fetch_column_metadata(with_column_lineage=True)
```
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:795]: We should also consider making `with_column_lineage` default to `True` in the backend, so users will have to opt-out instead. This can be a separate PR though.

**Review by rexledesma — CHANGES_REQUESTED** [review]
I don't think it's immediately obvious that you can chain these methods together (e.g. `.fetch_row_counts()`, `.fetch_column_metadata()`). Maybe showing the explicit chaining should be in its own section?

Also, this can happen on a separate PR as well, but we should also consider a section where we show how folk can add arbitrary metadata to the dbt event stream by using `.with_metadata(...)`.

File: docs/content/integrations/dbt/reference.mdx [hunk 3]
```diff
@@ -831,57 +830,55 @@ Once your dbt models have been materialized, you can view the row count data in
...
+    manifest=my_dbt_project.manifest_path,
+)
+def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    yield from (
+        dbt.cli(["build"], context=context).stream().fetch_column_metadata()
+    )
 ```
 
-Column dependencies can be removed from materialization metadata by disabling the collection of parent relation metadata. This can be done by setting the `enable_parent_relation_metadata_collection` argument to `False` in the `dagster.log_column_level_metadata()` macro:
+This method can be chained with other metadata fetching methods like <PyObject object="fetch_row_counts" module="dagster_dbt.core.resources_v2.DbtEventIterator" displayText="fetch_row_counts()" />:
```
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:863]: This section should have a header? "Composing the fetching of metadata"

**Review by rexledesma — APPROVED** [review]
